File: schemas/helpdesk-customer-support/drizzle/business_schedules.ts
// business_schedules: named business-hour schedules with timezone for SLA clock calculation.
// See README.md for full design rationale.

import { pgTable, uuid, text, boolean, timestamp } from "drizzle-orm/pg-core";

export const businessSchedules = pgTable(
  "business_schedules",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    timezone: text("timezone").notNull(),
    isDefault: boolean("is_default").notNull().default(false),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  }
);

File: schemas/helpdesk-customer-support/spacetimedb/business_schedules.rs
// business_schedules: Operating hour schedules for SLA calculation.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = business_schedules, public)]
pub struct BusinessSchedule {
    #[primary_key]
    pub id: String, // UUID
    pub name: String,
    pub timezone: String,
    pub is_default: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
